pkg/promise/promise_test.go

```diff
@@ -0,0 +1,28 @@
+package promise
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGo(t *testing.T) {
+	errCh := Go(functionWithError)
+	er := <-errCh
+	require.NotNil(t, er.Error())
+	require.EqualValues(t, "Error Occured", er.Error())
+	t.Log("GO promise with channel value 'Error' executed successfully")
+
+	noErrCh := Go(functionWithNoError)
+	er = <-noErrCh
+	require.Nil(t, er)
+	t.Log("GO promise with channel value 'No Error' executed successfully")
+}
+
+func functionWithError() (err error) {
+	return errors.New("Error Occured")
```

Comment:
Just noticed a typo... Occured should be Occurred (doesn't matter for the test, but might as well fix it).